sdcm/sct_runner.py

```diff
@@ -600,8 +602,67 @@ def _create_instance(self,
         LOGGER.info("Got public IP: %s", instance.public_ip_address)
         self.instance = instance
 
+        # If the SCT runner instance is used as a Docker backend for tests
+        if os.getenv("SCT_CLUSTER_BACKEND") == "docker":
```

Comment:
`self.params.get("cluster_backend")` must be used for it.
There is big enough probability to not have exactly env var be set, but yaml config option.

Reply:
The implemented logic is intended to be used only when Docker backend test run is started in Jenkins. In this case the env parameter is always set here https://github.com/scylladb/scylla-cluster-tests/blob/1f38a636ca1695c7bb0287dc9fd176dec7e192bf/vars/runSctTest.groovy#L28

And `self.params.get("cluster_backend")` won't help here. When running tests from Jenkins there is this differentiation between 'cluster backend for tests' and cloud provider for the 'cluster', like it is defined in https://github.com/scylladb/scylla-cluster-tests/blob/1f38a636ca1695c7bb0287dc9fd176dec7e192bf/vars/getCloudProviderFromBackend.groovy#L3

So, when the test is started from Jenkins, in the condition above we need to check what the selected backend for tests is (SCT_CLUSTER_BACKEND has this value), not the cloud provider (the value returned by `self.params.get("cluster_backend")`)

Reply:
@dimakr 
The `self.params.get("cluster_backend")` will always be equal to the `os.getenv("SCT_CLUSTER_BACKEND")` if the latter is set. And I proposed you to use the former approach with `self.params` to cover cases when env var is not set.
The `vars/getCloudProviderFromBackend.groovy` file is not related to the raised concern at all.

Reply:
Changes to use params for getting cluster_backend value.